requirements.txt

```diff
@@ -14,6 +14,7 @@ gitdb==4.0.10
 GitPython==3.1.35
 idna==3.4
 iniconfig==2.0.0
+ipdb==0.13.13
```

Comment:
I added it because it was used in some files, but I guess that was a temporary development thing and it probably shouldn't be in the requirements. I was thinking we should probably split the requirements into `requirements.txt` and `requirements-test.txt` as there are a lot of requirements that are only relevant for development: pytest*, ruff, pyright, black, isort and maybe others.